### ISSUE TITLE
NVTabular Docker image does not build

### ISSUE THREAD

**Issue by EnricoMi** [horovod/horovod#3691]
The `docker/horovod-nvtabular/Dockerfile` Docker image does not build in master any more:
https://github.com/horovod/horovod/runs/8261962616?check_suite_focus=true#step:8:2828
```
#27 40.96 Traceback (most recent call last):
#27 40.96   File "<string>", line 1, in <module>
#27 40.96   File "/root/miniconda3/lib/python3.8/site-packages/tensorflow/__init__.py", line 37, in <module>
#27 40.96     from tensorflow.python.tools import module_util as _module_util
#27 40.96   File "/root/miniconda3/lib/python3.8/site-packages/tensorflow/python/__init__.py", line 37, in <module>
#27 40.96     from tensorflow.python.eager import context
#27 40.96   File "/root/miniconda3/lib/python3.8/site-packages/tensorflow/python/eager/context.py", line 29, in <module>
#27 40.96     from tensorflow.core.framework import function_pb2
#27 40.96   File "/root/miniconda3/lib/python3.8/site-packages/tensorflow/core/framework/function_pb2.py", line 7, in <module>
#27 40.96     from google.protobuf import descriptor as _descriptor
#27 40.96   File "/root/miniconda3/lib/python3.8/site-packages/google/protobuf/descriptor.py", line 40, in <module>
#27 40.96     from google.protobuf.internal import api_implementation
#27 40.96   File "/root/miniconda3/lib/python3.8/site-packages/google/protobuf/internal/api_implementation.py", line 104, in <module>
#27 40.96     from google.protobuf.pyext import _message
#27 40.96 TypeError: bases must be types
```

It is likely that conflicting dependencies break tensorflow (e.g. #3684). Try to compare versions installed earlier when the image used to build with versions now being installed.

**Comment by EnricoMi** [comment]
@leewyang the Docker image added in #3603 stopped building. Can you look into this, please?

**Comment by maxhgerlach** [comment]
This appeared to be fixed via #3694, but a similar error seems to have crept up on master again, see for instance: https://github.com/horovod/horovod/actions/runs/3047355141/jobs/4923088862

```
#27 31.47 ERROR: pip's dependency resolver does not currently take into account all the packages that are installed. This behaviour is the source of the following dependency conflicts.
#27 31.47 dask-cudf 21.10.1 requires cupy-cuda112, which is not installed.
#27 31.47 cudf 21.10.1 requires cupy-cuda112, which is not installed.
#27 31.47 cudf 21.10.1 requires Cython<0.30,>=0.29, which is not installed.
#27 31.47 nvtabular 0.9.0 requires pandas>=1.2.0, but you have pandas 1.0.5 which is incompatible.
#27 31.47 Successfully installed keras-2.8.0 pandas-1.0.5 scipy-1.9.1
#27 31.64 WARNING: You are using pip version 21.0.1; however, version 22.2.2 is available.
#27 31.64 You should consider upgrading via the '/root/miniconda3/bin/python -m pip install --upgrade pip' command.
#27 32.25 Traceback (most recent call last):
#27 32.25   File "<string>", line 1, in <module>
#27 32.25   File "/root/miniconda3/lib/python3.8/site-packages/tensorflow/__init__.py", line 37, in <module>
#27 32.25     from tensorflow.python.tools import module_util as _module_util
#27 32.25   File "/root/miniconda3/lib/python3.8/site-packages/tensorflow/python/__init__.py", line 37, in <module>
#27 32.25     from tensorflow.python.eager import context
#27 32.25   File "/root/miniconda3/lib/python3.8/site-packages/tensorflow/python/eager/context.py", line 29, in <module>
#27 32.25     from tensorflow.core.framework import function_pb2
#27 32.25   File "/root/miniconda3/lib/python3.8/site-packages/tensorflow/core/framework/function_pb2.py", line 7, in <module>
#27 32.25     from google.protobuf import descriptor as _descriptor
#27 32.25   File "/root/miniconda3/lib/python3.8/site-packages/google/protobuf/descriptor.py", line 47, in <module>
#27 32.25     from google.protobuf.pyext import _message
#27 32.25 TypeError: bases must be types
#27 ERROR: process "/bin/bash -euo pipefail -c if [[ ${TENSORFLOW_PACKAGE} != \"tf-nightly-gpu\" ]]; then         pip install --no-cache-dir ${TENSORFLOW_PACKAGE} \"protobuf<4\";         if [[ ${KERAS_PACKAGE} != \"None\" ]]; then             pip uninstall -y keras;             pip install --no-cache-dir ${KERAS_PACKAGE} \"scipy!=1.4.0\" \"pandas<1.1.0\";         fi;         mkdir -p ~/.keras;         ldconfig /usr/local/cuda/targets/x86_64-linux/lib/stubs;         python -c \"import tensorflow as tf; tf.keras.datasets.mnist.load_data()\";         ldconfig;     fi" did not complete successfully: exit code: 1
------
 > [22/36] RUN if [[ tensorflow-gpu==2.8.0 != "tf-nightly-gpu" ]]; then         pip install --no-cache-dir tensorflow-gpu==2.8.0 "protobuf<4";         if [[ keras==2.8.0 != "None" ]]; then             pip uninstall -y keras;             pip install --no-cache-dir keras==2.8.0 "scipy!=1.4.0" "pandas<1.1.0";         fi;         mkdir -p ~/.keras;         ldconfig /usr/local/cuda/targets/x86_64-linux/lib/stubs;         python -c "import tensorflow as tf; tf.keras.datasets.mnist.load_data()";         ldconfig;     fi:
#27 32.25     from tensorflow.python.tools import module_util as _module_util
#27 32.25   File "/root/miniconda3/lib/python3.8/site-packages/tensorflow/python/__init__.py", line 37, in <module>
#27 32.25     from tensorflow.python.eager import context
#27 32.25   File "/root/miniconda3/lib/python3.8/site-packages/tensorflow/python/eager/context.py", line 29, in <module>
#27 32.25     from tensorflow.core.framework import function_pb2
#27 32.25   File "/root/miniconda3/lib/python3.8/site-packages/tensorflow/core/framework/function_pb2.py", line 7, in <module>
#27 32.25     from google.protobuf import descriptor as _descriptor
#27 32.25   File "/root/miniconda3/lib/python3.8/site-packages/google/protobuf/descriptor.py", line 47, in <module>
#27 32.25     from google.protobuf.pyext import _message
#27 32.25 TypeError: bases must be types
------
Dockerfile:132
--------------------
 131 |     # Pin protobuf<4 for tensorflow: https://github.com/tensorflow/tensorflow/issues/56815
 132 | >>> RUN if [[ ${TENSORFLOW_PACKAGE} != "tf-nightly-gpu" ]]; then \
 133 | >>>         pip install --no-cache-dir ${TENSORFLOW_PACKAGE} "protobuf<4"; \
 134 | >>>         if [[ ${KERAS_PACKAGE} != "None" ]]; then \
 135 | >>>             pip uninstall -y keras; \
 136 | >>>             pip install --no-cache-dir ${KERAS_PACKAGE} "scipy!=1.4.0" "pandas<1.1.0"; \
 137 | >>>         fi; \
 138 | >>>         mkdir -p ~/.keras; \
 139 | >>>         ldconfig /usr/local/cuda/targets/x86_64-linux/lib/stubs; \
 140 | >>>         python -c "import tensorflow as tf; tf.keras.datasets.mnist.load_data()"; \
 141 | >>>         ldconfig; \
 142 | >>>     fi
 143 |     
--------------------
ERROR: failed to solve: process "/bin/bash -euo pipefail -c if [[ ${TENSORFLOW_PACKAGE} != \"tf-nightly-gpu\" ]]; then         pip install --no-cache-dir ${TENSORFLOW_PACKAGE} \"protobuf<4\";         if [[ ${KERAS_PACKAGE} != \"None\" ]]; then             pip uninstall -y keras;             pip install --no-cache-dir ${KERAS_PACKAGE} \"scipy!=1.4.0\" \"pandas<1.1.0\";         fi;         mkdir -p ~/.keras;         ldconfig /usr/local/cuda/targets/x86_64-linux/lib/stubs;         python -c \"import tensorflow as tf; tf.keras.datasets.mnist.load_data()\";         ldconfig;     fi" did not complete successfully: exit code: 1
Error: buildx failed with: ERROR: failed to solve: process "/bin/bash -euo pipefail -c if [[ ${TENSORFLOW_PACKAGE} != \"tf-nightly-gpu\" ]]; then         pip install --no-cache-dir ${TENSORFLOW_PACKAGE} \"protobuf<4\";         if [[ ${KERAS_PACKAGE} != \"None\" ]]; then             pip uninstall -y keras;             pip install --no-cache-dir ${KERAS_PACKAGE} \"scipy!=1.4.0\" \"pandas<1.1.0\";         fi;         mkdir -p ~/.keras;         ldconfig /usr/local/cuda/targets/x86_64-linux/lib/stubs;         python -c \"import tensorflow as tf; tf.keras.datasets.mnist.load_data()\";         ldconfig;     fi" did not complete successfully: exit code: 1
```
Would you take another look @leewyang ?

**Comment by leewyang** [comment]
Yeah, this Dockerfile doesn't actually involve NVTabular at all, so I'm fairly certain it's unrelated and just a standard issue with changing dependencies.  Anyhow, I've pushed a PR to apply the same fix to both of the Dockerfile.test.* files.  Hoping that's sufficient... but it's hard to tell without going thru the full build.

**Comment by maxhgerlach** [comment]
Apparently the problem has "resolved itself", see comments in #3701. Thanks for giving this another shot anyhow.
